src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -66,13 +73,19 @@ def handleRequest(request)
   end
 
   def compileCatalog(request_data)
-    @catalog_compiler.compile(convert_java_args_to_ruby(request_data))
+    Puppet.override(lookup_key_recorder: create_recorder) do
```

Comment:
I was trying to think if triggering collection on this endpoint when it's used with impact analysis would get weird (e.g. overwrite inputs from real catalogs), but currently we're not saving any of the catalogs compiled that way in PDB, so I think this is fine. It will collect for ACE nodes, but that's probably a good thing.